appmap/_implementation/generation.py

```diff
@@ -114,5 +115,8 @@ def default(self, o):
 
 
 def dump(recording, metadata=None, indent=None):
+    """Generate a JSON representation of an appmap."""
+    if hasattr(recording, 'events'):
+        recording = recording.events
```

Comment:
I can't, this will break https://github.com/applandinc/appmap-python/pull/98/files#diff-06d67252592021259f8167f0ef2fc8970192a7aa947478e8d00d210e19067e3aR162 -- this is exactly why I've made this change.

Reply:
I'm confused. Doesn't that line you linked to return a recording that has an `events` attribute? When is `dump` called with with an object that doesn't have an `events` attribute?

Reply:
No, it returns event list directly, cf. https://github.com/applandinc/appmap-python/blob/879ee909fcb60b0d7527fc86d62500a90b627c81/appmap/_implementation/recording.py#L232

Reply:
I still think the control coupling in this interface is bad, but I'm tired of fighting about it.